### PR TITLE
Make OtlpMeterRegistry virtual thread aware

### PR DESCRIPTION
**Pull request by mturbe**
Add support for virtual threads in OtlpMeterRegistry configuration per issue #42175.

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpProperties.java [hunk 1]
```diff
@@ -79,6 +79,11 @@ public class OtlpProperties extends StepRegistryProperties {
 	 */
 	private TimeUnit baseTimeUnit = TimeUnit.MILLISECONDS;
 
+	/**
+	 * Whether virtual threads should be used for publishing metrics.
+	 */
+	private boolean virtualThreadsEnabled = false;
```
Comment (mhalbritter) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpProperties.java:85]: We don't need a new property here. We have a condition, named `ConditionalOnThreading` which can be used to detect if the user wants to enable virtual threads. There are already usages in the codebase for this annotation which you can use to see how it's used.

**Review by mhalbritter — CHANGES_REQUESTED** [review]
Hello @mturbe, thanks for the PR. I've added a comment for your consideration.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpMetricsExportAutoConfigurationTests.java [hunk 2]
```diff
@@ -76,6 +79,35 @@ void allowsCustomConfigToBeUsed() {
...
+			assertThat(properties.isVirtualThreadsEnabled()).isFalse();
+			OtlpMeterRegistry registry = context.getBean(OtlpMeterRegistry.class);
+			assertThat(registry).extracting("scheduledExecutorService")
+				.satisfies((executor) -> ScheduledExecutorServiceAssert.assertThat((ScheduledExecutorService) executor)
+					.usesPlatformThreads());
+		});
+	}
+
+	@Test
+	void allowsVirtualThreadsToBeUsed() {
```
Comment (mhalbritter) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpMetricsExportAutoConfigurationTests.java:96]: We need to guard this test to only run on Java 21 and up.

File: spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/assertj/ScheduledExecutorServiceAssert.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		isNotNull();
+		if (!producesVirtualThreads()) {
+			failWithMessage("Expected executor to use virtual threads, but it uses platform threads");
+		}
+		return this;
+	}
+
+	private boolean producesVirtualThreads() {
+		try {
+			return this.actual.schedule(() -> Thread.currentThread().isVirtual(), 0, TimeUnit.SECONDS).get();
```
Comment (mhalbritter) [inline spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/assertj/ScheduledExecutorServiceAssert.java:68]: We can't use this API, as our codebase is on Java 17, and `isVirtual` has been (stable) introduced with Java 21.

**Review by mhalbritter — CHANGES_REQUESTED** [review]

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpProperties.java (hunk 1 above)
Comment (mturbe) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpProperties.java:85]: Great, thanks for sharing that. I'll switch to `ConditionalOnThreading`.

**Comment by mhalbritter** [comment]
Thank you very much and congratulations on your first contribution :tada:!

**Pull request closed without merging**